### PR TITLE
[Merged by Bors] - Fix clippy issue for benches crate

### PR DESCRIPTION
**Pull request by lewiszlw**
# Objective

- https://github.com/bevyengine/bevy/pull/3505 marked `S-Adopt-Me` , this pr is to continue his work.

## Solution

- run `cargo clippy --workspace --all-targets --all-features -- -Aclippy::type_complexity -Wclippy::doc_markdown -Wclippy::redundant_else -Wclippy::match_same_arms -Wclippy::semicolon_if_nothing_returned -Wclippy::explicit_iter_loop -Wclippy::map_flatten -Dwarnings` under benches dir.
- fix issue according to suggestion.

### CODE REVIEW

**Review by james7132 — APPROVED** [review]

File: benches/benches/bevy_reflect/map.rs [hunk 1]
```diff
@@ -262,11 +262,12 @@ fn dynamic_map_insert(criterion: &mut Criterion) {
...
             |bencher, &size| {
                 bencher.iter_batched(
-                    || DynamicMap::default(),
+                    DynamicMap::default,
                     |mut map| {
                         for i in 0..size as u64 {
                             let key = black_box(i);
-                            black_box(map.insert(key, i));
+                            map.insert(key, i);
+                            black_box(());
```
Comment (hymm) [inline benches/benches/bevy_reflect/map.rs:270]: this is a no-op. You should just delete the black_box.

File: benches/benches/bevy_reflect/struct.rs [hunk 2]
```diff
@@ -191,7 +191,8 @@ fn dynamic_struct_insert(criterion: &mut Criterion) {
                 bencher.iter_batched(
                     || s.clone_dynamic(),
                     |mut s| {
-                        black_box(s.insert(black_box(&field), ()));
+                        s.insert(black_box(&field), ());
+                        black_box(());
```
Comment (hymm) [inline benches/benches/bevy_reflect/struct.rs:195]: this is a no-op. You should just delete the black_box

**Comment by alice-i-cecile** [comment]
This PR is mergeable in its current form, but the comments raised are valid and it'll be easier to do it here. @lewiszlw want to tackle it, or would you prefer I just merge?

**Comment by lewiszlw** [comment]
Resolved comments. @alice-i-cecile

File: benches/benches/bevy_reflect/map.rs [hunk 3]
```diff
@@ -262,11 +262,11 @@ fn dynamic_map_insert(criterion: &mut Criterion) {
             &size,
             |bencher, &size| {
                 bencher.iter_batched(
-                    || DynamicMap::default(),
+                    DynamicMap::default,
                     |mut map| {
                         for i in 0..size as u64 {
                             let key = black_box(i);
-                            black_box(map.insert(key, i));
+                            map.insert(key, i);
```
Comment (mockersf) [inline benches/benches/bevy_reflect/map.rs:269]: should map be blackboxed?

Comment (lewiszlw) [reply benches/benches/bevy_reflect/map.rs:269]: black_box is a no-op according to hymm's comments, it should be deleted.

File: benches/benches/bevy_reflect/struct.rs [hunk 4]
```diff
@@ -191,7 +191,7 @@ fn dynamic_struct_insert(criterion: &mut Criterion) {
                 bencher.iter_batched(
                     || s.clone_dynamic(),
                     |mut s| {
-                        black_box(s.insert(black_box(&field), ()));
+                        s.insert(black_box(&field), ());
```
[the remark this answers is not in the document]
Comment (lewiszlw) [reply benches/benches/bevy_reflect/struct.rs:194]: black_box is a no-op according to hymm's comments, it should be deleted.

File: benches/benches/bevy_reflect/map.rs (hunk 3 above)
Comment (mockersf) [reply benches/benches/bevy_reflect/map.rs:269]: `black_box(());` is a no-op, not sure about `black_box(map)` in this case

File: benches/benches/bevy_reflect/struct.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (mockersf) [reply benches/benches/bevy_reflect/struct.rs:194]: `black_box(());` is a no-op, not sure about `black_box(s)` in this case

File: benches/benches/bevy_reflect/map.rs (hunk 3 above)
Comment (lewiszlw) [reply benches/benches/bevy_reflect/map.rs:269]: Sorry I misunderstood.

**Comment by alice-i-cecile** [comment]
Waiting for more clarity on the points raised above.

**Comment by alice-i-cecile** [comment]
@lewiszlw could you re-add those black box calls to this PR so we can merge it and investigate the matter elsewhere?

**Comment by lewiszlw** [comment]
Re-added. Do I need to create an issue to track this?

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
> Re-added. Do I need to create an issue to track this?

That would be lovely :)

bors r+

**Pull request closed without merging**